src/seccompiler-bin/src/main.rs

```diff
@@ -0,0 +1,456 @@
+//! and easily install them on a per-thread basis, in order to achieve a quick and robust
+//! seccomp-based jailing solution.
+//!
+//! See the documentation on github for more information.
+//!
+//!  ```text
+//! The compilation goes through a couple of steps, from JSON to BPF:
+//!
+//!                  JSON
+//!                   |
+//!            (via serde_json)
+//!                   |
+//!                   V
+//!       collection of `Filter` objects
+//!                   |
+//!      (via Compiler.compile_blob(...))
+//!                   |
+//!                   V
+//!   collection of `SeccompFilter` objects
+//!     (IR - intermediate representation)
+//!                   |
+//!    (via SeccompFilter.try_into::<BpfProgram>(...))
+//!                   |
+//!                   V
+//!     collection of `BpfProgram` objects
+//! ```
+
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::{fmt, io, process};
+
+use bincode::Error as BincodeError;
+use seccompiler::{compile_from_json, Error as SeccompilerError, TargetArch};
+use utils::arg_parser::{ArgParser, Argument, Arguments as ArgumentsBag};
+use utils::seccomp::sock_filter;
+
+const SECCOMPILER_VERSION: &str = env!("FIRECRACKER_VERSION");
+const DEFAULT_OUTPUT_FILENAME: &str = "seccomp_binary_filter.out";
+const EXIT_CODE_ERROR: i32 = 1;
+
+#[derive(Debug)]
+enum Error {
+    Bincode(BincodeError),
+    FileOpen(PathBuf, io::Error),
+    MissingInputFile,
+    MissingTargetArch,
+    Seccompiler(SeccompilerError),
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        match *self {
+            Bincode(ref err) => write!(f, "Bincode (de)serialization failed: {}", err),
+            FileOpen(ref path, ref err) => write!(
+                f,
+                "{}",
+                format!("Failed to open file {:?}: {}", path, err).replace("\"", "")
+            ),
+            MissingInputFile => write!(f, "Missing input file."),
+            MissingTargetArch => write!(f, "Missing target arch."),
+            Seccompiler(ref err) => write!(f, "Seccompiler error: {}", err),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+struct Arguments {
+    input_file: String,
+    output_file: String,
+    target_arch: TargetArch,
+}
+
+fn build_arg_parser() -> ArgParser<'static> {
+    ArgParser::new()
+        .arg(
+            Argument::new("input-file")
+                .required(true)
+                .takes_value(true)
+                .help("File path of the JSON input."),
+        )
+        .arg(
+            Argument::new("output-file")
+                .required(false)
+                .takes_value(true)
+                .default_value(DEFAULT_OUTPUT_FILENAME)
+                .help("Optional path of the output file."),
+        )
+        .arg(
```

Comment:
just an idea but what if we would use a default value for this as well. DEFAULT_TARGET which is the current target and we define the variable using target_arch = "..."

Reply:
I wouldn't do this. Users may then become unaware that these filters are arch-dependent and may overlook this, resulting in catastrophic runtime errors when executing the filter.